bip-0322.mediawiki

```diff
@@ -76,16 +88,33 @@ The <code>to_sign</code> transaction is:
 
 A full signature consists of the base64-encoding of the <code>to_sign</code> transaction in standard network serialisation once it has been signed.
 
+This signature, as-is, only proves a message is signed by the intended recipient of the given invoice address which creates the <code>message_challenge</code> script.
+It does not prove any control of bitcoins or having had sent any transactions.
+
+TODO: How does this interact with as-of-yet-unspecified "Silent Transactions"?
+
+TODO: Some invalid opcode to allow only in various proof types?
```

Comment:
This seems like it should be too trivial to put off as "future work". IMO.

Reply:
@luke-jr Here's a solution at least for the Silent Transactions. Since they can't decide which address formats they will use for it (there is a lot of to-and-fro between taproot and segwit), and all such address formats can be represented in the Simple format anyway, just write in this BIP that Silent Transactions can use the Simple format to sign their messages, and the scriptPubKey/scriptSig/Witness will all be the same.

Reply:
Making an invalid opcode valid would be useful to provide a separate Tapscript that is valid _only_ for signing messages, but not actual spending.

Reply:
If someone wants to hide the specific allocation of outgpong value among tapscript outputs, then it would already be possible to create a branch that spends the UTXO to an OP_RETURN.  Since BIP322-signature-txs cannot be spent, there's no downside to signing that branch.

Reply:
Note that we can also redefine the meaning of OP_CHECKSIG in the context of message signing and verifying to hash the virtual transaction that is derived from the message i.e. `to_spend`. [Bitcointalk source](https://bitcointalk.org/index.php?topic=5408898.msg60728854#msg60728854)